models/DataObject/Fieldcollection/Definition.php

```diff
@@ -305,7 +305,7 @@ public function isForbiddenName(): bool
         if ($key === null || $key === '') {
             return true;
         }
-        if (!preg_match('/^[a-zA-Z][a-zA-Z0-9]*$/', $key)) {
+        if (!preg_match('/^[a-zA-Z][a-zA-Z0-9_]*$/', $key)) {
```

Comment:
Shouldn't it be `\w` here as well?

```suggestion
        if (!preg_match('/^[a-zA-Z]\w*$/', $key)) {
```